iaso/api/mobile/entity.py

```diff
@@ -0,0 +1,106 @@
+from django.db.models import Prefetch
+from django_filters.rest_framework import DjangoFilterBackend  # type: ignore
+from rest_framework import filters
+from rest_framework import serializers
+from rest_framework.pagination import PageNumberPagination
+
+from iaso.api.common import (
+    ModelViewSet,
+    DeletionFilterBackend,
+    TimestampField,
+)
+from iaso.models import Entity, Instance, OrgUnit
+
+
+class LargeResultsSetPagination(PageNumberPagination):
+    page_size = 1000
+    page_size_query_param = "page_size"
+    max_page_size = 1000
+
+
+class MobileEntityAttributesSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Instance
+        fields = ["id", "form_id", "form_version_id", "created_at", "updated_at", "org_unit_id", "json"]
+
+    form_id = serializers.IntegerField(read_only=True, source="id")
+    id = serializers.CharField(read_only=True, source="uuid")
+    form_version_id = serializers.CharField(read_only=True, source="form.form.id")
+    org_unit_id = serializers.CharField(read_only=True, source="org_unit.id")
+
+    created_at = TimestampField()
+    updated_at = TimestampField()
+
+
+class MobileEntitySerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Entity
+        fields = [
+            "id",
+            "created_at",
+            "updated_at",
+            "defining_instance_id",
+            "entity_type_id",
+            "instances",
+        ]
+
+    created_at = TimestampField()
+    updated_at = TimestampField()
+
+    instances = serializers.SerializerMethodField()
+    id = serializers.CharField(read_only=True, source="uuid")
+    defining_instance_id = serializers.CharField(read_only=True, source="attributes.uuid")
+    entity_type_id = serializers.CharField(read_only=True, source="entity_type.id")
+
+    @staticmethod
+    def get_instances(entity: Entity):
+        res = []
+        for instance in entity.non_deleted_instances:  # type: ignore
+            res.append(MobileEntityAttributesSerializer(instance).data)
+
+        return res
+
+    @staticmethod
+    def get_entity_type_name(obj: Entity):
+        return obj.entity_type.name if obj.entity_type else None
+
+
+class MobileEntityViewSet(ModelViewSet):
+    """Entity API for mobile
+
+    list: /api/mobile/entity
+
+    pagination by default: 1000 entities
+
+    It's possible to filter out entities with no activity before a certain date with the parameter limit_date
+
+    details = /api/mobile/entity/id
```

Comment:
I believe that the default is to use the "id" and not the "uuid", therefore, this endpoint is not correct. The "id" field is an implementation detail of the backend and should not be exposed.